cmd/eskip/write.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2015 Zalando SE
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"github.com/zalando/skipper/eskip"
+	etcdclient "github.com/zalando/skipper/etcd"
+	"github.com/zalando/skipper/filters/flowid"
+	"regexp"
+)
+
+const randomIdLength = 16
+
+type (
+	routePredicate func(*eskip.Route) bool
+	routeMap       map[string]*eskip.Route
+)
+
+var routeIdRx = regexp.MustCompile("\\W")
+
+func any(_ *eskip.Route) bool { return true }
+
+func routesDiffer(left, right *eskip.Route) bool {
+	return left.String() != right.String()
+}
+
+func mapRoutes(routes routeList) routeMap {
+	m := make(routeMap)
+	for _, r := range routes {
+		m[r.Id] = r
+	}
+
+	return m
+}
+
+// generate weak random id for a route if
+// it doesn't have one.
+func ensureId(r *eskip.Route) error {
+	if r.Id != "" {
+		return nil
+	}
+
+	id, err := flowid.NewFlowId(randomIdLength)
+	if err != nil {
+		return err
+	}
+
+	// replace characters that are not allowed
+	// for eskip route ids.
+	id = routeIdRx.ReplaceAllString(id, "x")
```

Comment:
I didn't want to add an additional dependency, and considered the randomness of this solution good enough for the purpose.